sdk/core/core/src/az_http_request.c

```diff
@@ -26,9 +26,9 @@ AZ_NODISCARD az_result az_http_request_init(
     az_span body)
 {
   AZ_PRECONDITION_NOT_NULL(p_hrb);
-  AZ_PRECONDITION_VALID_SPAN(method, 0);
-  AZ_PRECONDITION_VALID_SPAN(url, 0);
-  AZ_PRECONDITION_VALID_SPAN(headers_buffer, 0);
+  AZ_PRECONDITION_VALID_SPAN(method, 0, false);
```

Comment:
Now that I look at these again, the min_length for method & url should be 1; headers can be 0 - please fix